src/customHooks/useYoutube.js

```diff
@@ -0,0 +1,110 @@
+                        q: searchTerm,
+                        type: 'video'
+                    }
+                }).then(res => {
+                    setSearchRestult(res.data);
+                }).catch(err => {
+                    console.error(err);
+                    setSearchError(err);
+                });
+
+            setSearchIsLoading(false);
+        }
+
+        fetchData();
+    }, [searchTerm, searchMaxResults]);
+
+    return { searchResult, searchIsLoading, searchError, setSearchTerm, setSearchMaxResults };
+}
+
+export function useYoutubeVideo() {
+    const [videoId, setVideoId] = useState(null);
+    const [videoIsLoading, setVideoIsLoading] = useState(true);
+    const [videoError, setVideoError] = useState(null);
+    const [videoData, setVideoData] = useState(null);
+
+    useEffect(() => {
+
+        const fetchData = async () => {
+            setVideoIsLoading(true);
+
+            await axios.get('https://youtube.googleapis.com/youtube/v3/videos', {
+                params: {
+                    part: 'snippet,player',
+                    id: videoId,
+                    key: process.env.REACT_APP_YOUTUBE_API_KEY
+                }
+            }).then(res => {
+                setVideoData(res.data);
+            }).catch(err => {
+                console.error(err);
+                setVideoError(err);
+            });
+
+            setVideoIsLoading(false);
+        }
+
+        videoId && fetchData();
+
+    }, [videoId]);
+
+    return { videoData, videoIsLoading, videoError, videoId, setVideoId };
+
+}
+
+export function useYoutubeRelatedVideos(videoId, defaultMaxResults) {
+    const [relatedVideosIsLoading, setRelatedVideosIsLoading] = useState(true);
+    const [relatedVideosResult, setRelatedVideosResult] = useState(null);
+    const [relatedVideosError, setRelatedVideosError] = useState(null);
+    const [maxRelatedVideosResults, setMaxRelatedVideosResults] = useState(defaultMaxResults);
+
+    useEffect(() => {
+        const fetchData = async () => {
+            setRelatedVideosIsLoading(true);
+
+            await axios.get('https://www.googleapis.com/youtube/v3/search',
+                {
+                    params: {
+                        part: 'snippet',
+                        maxResults: maxRelatedVideosResults,
+                        key: process.env.REACT_APP_YOUTUBE_API_KEY,
+                        relatedToVideoId: videoId,
+                        type: 'video'
+                    }
+                }).then(res => {
+                    setRelatedVideosResult(res.data);
+                }).catch(err => {
+                    console.error(err);
+                    setRelatedVideosError(err);
+                });
+
+            setRelatedVideosIsLoading(false);
+        }
+
+        videoId && fetchData();
+
+    }, [videoId, maxRelatedVideosResults]);
+
+    return { relatedVideosResult, relatedVideosIsLoading, relatedVideosError, setMaxRelatedVideosResults };
+}
```

Comment:
creo que cada hook debería vivir en su propio archivo independiente

Reply:
Cierto, noté que es una práctica común tener un archivo por hook, pero en realidad no entiendo bien por qué. Hay alguna razón técnica para tenerlos separados o es mera convención? Igual los separé cada uno en su propio archivo y carpeta junto con sus tests. En realidad sí me parece más conveniente en cuestión de organización 🙌 este cambio fue echo en el commit 84ee253

Reply:
desconozco si sea alguna convención, pero yo lo prefiero de esa manera para mantener un orden y sea más fácil encontrarlos en dado caso de que quiera utilizarlos o modificarlos.